src/hooks.ts

```diff
@@ -16,24 +16,27 @@ export function useCallsDataSubscription(
   context: MulticallContext,
   chainId: number | undefined,
   calls: Array<Call | undefined>,
-  blocksPerFetch = 1
+  blocksPerFetch?: number
 ): CallResult[] {
   const { reducerPath, actions } = context
   const callResults = useSelector((state: WithMulticallState) => state[reducerPath].callResults)
+  const defaultListenerOptions = useSelector((state: WithMulticallState) => state[reducerPath].defaultListenerOptions)
   const dispatch = useDispatch()
-
   const serializedCallKeys: string = useMemo(() => JSON.stringify(callsToCallKeys(calls)), [calls])
 
   // update listeners when there is an actual change that persists for at least 100ms
   useEffect(() => {
     const callKeys: string[] = JSON.parse(serializedCallKeys)
     const calls = callKeysToCalls(callKeys)
     if (!chainId || !calls) return
+    const blocksPerFetchForChain =
+      blocksPerFetch ?? (defaultListenerOptions ?? {})[chainId]?.blocksPerFetch ?? DEFAULT_BLOCKS_PER_FETCH
```

Comment:
can you unpack this? the logic is nested enough that is starting to get hard to parse

Reply:
good call, done